libraries/plugins/apis/account_history_api/account_history_api.cpp

```diff
@@ -1,92 +1,212 @@
+}
+
+class account_history_api_rocksdb_impl : public abstract_account_history_api_impl
+{
+   public:
+      account_history_api_rocksdb_impl() :
+         abstract_account_history_api_impl(), _dataSource( appbase::app().get_plugin< steem::plugins::rocksdb::rocksdb_plugin >() ) {}
+      ~account_history_api_rocksdb_impl() {}
+
+      get_ops_in_block_return get_ops_in_block( const get_ops_in_block_args& ) override;
+      get_transaction_return get_transaction( const get_transaction_args& ) override;
+      get_account_history_return get_account_history( const get_account_history_args& ) override;
+      enum_virtual_ops_return enum_virtual_ops( const enum_virtual_ops_args& ) override;
+
+      const rocksdb::rocksdb_plugin& _dataSource;
+};
+
+DEFINE_API_IMPL( account_history_api_rocksdb_impl, get_ops_in_block )
+{
+   get_ops_in_block_return result;
+   _dataSource.find_operations_by_block(args.block_num,
+      [&result, &args](const rocksdb::rocksdb_operation_object& op)
+      {
+         api_operation_object temp(op);
+         if( !args.only_virtual || is_virtual_operation( temp.op ) )
+            result.ops.emplace_back(std::move(temp));
+      }
+   );
+   return result;
+}
+
+DEFINE_API_IMPL( account_history_api_rocksdb_impl, get_account_history )
+{
+   FC_ASSERT( args.limit <= 10000, "limit of ${l} is greater than maxmimum allowed", ("l",args.limit) );
+   FC_ASSERT( args.start >= args.limit, "start must be greater than limit" );
 
    get_account_history_return result;
-   while( itr != end )
-   {
-      result.history[ itr->sequence ] = _db.get( itr->op );
-      ++itr;
-   }
+
+   _dataSource.find_account_history_data(args.account, args.start, args.limit,
+      [&result, this](unsigned int sequence, const rocksdb::rocksdb_operation_object& op)
+      {
+         result.history[sequence] = api_operation_object(op);
+      });
+
+   return result;
+}
+
+DEFINE_API_IMPL( account_history_api_rocksdb_impl, get_transaction )
+{
+   FC_ASSERT( false, "This API is not supported for account history backed by RocksDB" );
+}
+
+DEFINE_API_IMPL( account_history_api_rocksdb_impl, enum_virtual_ops)
+{
+   enum_virtual_ops_return result;
+
+   result.next_block_range_begin = _dataSource.enum_operations_from_block_range(args.block_range_begin,
+      args.block_range_end,
+      [&result](const rocksdb::rocksdb_operation_object& op)
+      {
+         result.ops.emplace_back(api_operation_object(op));
+      }
+   );
 
    return result;
 }
 
 } // detail
 
-account_history_api::account_history_api(): my( new detail::account_history_api_impl() )
+account_history_api::account_history_api()
 {
+   auto ah_cb = appbase::app().find_plugin< steem::plugins::account_history::account_history_plugin >();
+   auto ah_rocks = appbase::app().find_plugin< steem::plugins::rocksdb::rocksdb_plugin >();
+
+   if( ah_rocks != nullptr )
+   {
+      if( ah_cb != nullptr )
```

Comment:
I think better could be add dedicated option `account-history-legacy-storage` which by default is false and allows to enable old AH implementation. It could be more readable to customers/admins than implicit decision caused by adding old account_history_plugin to the list.

Reply:
The point of this is that it is automatic. The option `account-history-legacy-storage` accomplishes the same thing they already have to do with `plugin=account_history` or `plugin=account_history_rocksdb`